src/framework/mpas_string_utils.F

```diff
@@ -0,0 +1,54 @@
+! Copyright (c) 2023 The University Corporation for Atmospheric Research (UCAR).
+!
+! Unless noted otherwise source code is licensed under the BSD license.
+! Additional copyright and license information can be found in the LICENSE file
+! distributed with this code, or at https://mpas-dev.github.io/license.html .
+!
+!-----------------------------------------------------------------------
+!  mpas_string_utils
+!
+!> \brief Collection of functions used for string manipulation
+!> \author Matthew Dimond
+!> \date   25 July 2023
+!> \details
+!>  This module provides functions and subroutines used for string
+!>  manipulations and utilities.
+!
+!-----------------------------------------------------------------------
+module mpas_string_utils
+
+  contains
+
+  subroutine mpas_split_string(string, delimiter, subStrings)
+
+    implicit none
+
+    character(len=*), intent(in) :: string
+    character, intent(in) :: delimiter
+    character(len=*), pointer, dimension(:) :: subStrings
+
+    integer :: i, start, index
+
+    index = 1
+    do i = 1, len(string)
+      if(string(i:i) == delimiter) then
+        index = index + 1
+      end if
+    end do
+
+    allocate(subStrings(1:index))
+
+    start = 1
+    index = 1
+    do i = 1, len(string)
+      if (string(i:i) == delimiter) then
+          subStrings(index) = string(start:i-1)
+          index = index + 1
+          start = i + 1
+      end if
+    end do
+    subStrings(index) = string(start:len(string))
+  end subroutine mpas_split_string
```

Comment:
It's a minor detail, but leaving a blank line before the `end subroutine` line would follow existing style.